retrofit/src/main/java/retrofit/RxSupport.java

```diff
@@ -21,24 +24,35 @@
   RxSupport() {
   }
 
-  Observable createRequestObservable(final Invoker invoker) {
-    return Observable.create(new Observable.OnSubscribe<Object>() {
-      @Override public void call(final Subscriber<? super Object> subscriber) {
-        invoker.invoke(new Invoker.Callback() {
-          @Override public void next(Object o) {
-            if (!subscriber.isUnsubscribed()) {
-              subscriber.onNext(o);
-              subscriber.onCompleted();
-            }
-          }
+  Observable createRequestObservable(final Invoker invoker, final RequestBuilder requestBuilder,
+      final RequestInterceptor requestInterceptor) {
+    return Observable.defer(new Func0<Observable<Object>>() {
```

Comment:
Either way would work. If you prefer I do it in `OnSubscribe` I don't mind. I tend to avoid using `Observable.create` when there is an operator available. In this case after digging into the `defer` code, it looks like it doesn't do any unsubscribe checking when returning the new subscription, so it may not be doing much benefit. I think if I move it in to the create, I should also add an isUnsubscribed check after the intercept and before the invoke to prevent the work launching if it's been unsubscribed (and since some of the interceptors, like for auth tokens, can take a round trip, it seems like a reasonable case.

If we wanted to go full operator and let Rx handle it all for us, one option would be to remove the callback invoker interface entirely and do a synchronous invocation inside an Observable, like this:

    Observable createRequestObservable(final RequestBuilder requestBuilder, final RequestInterceptor requestInterceptor,
        final OkHttpClient client, final MethodInfo methodInfo) {
      return Observable.defer(() -> {
        final Request request = requestInterceptor.intercept(requestBuilder);
        try {
          client.newCall(request).execute();
        } catch (final IOException e) {
          return Observable.error(RetrofitError.networkError(request.urlString(), e);
        }
        // This assumes we make createResult accessible from here.
        return Observable.just(createResult(methodInfo, response);
      });
    }

The advantages of this are that it's simpler codewise, and we no longer have to create and maintain our own observable. Also, clients have more control over where their network calls are executed on via subscribeOn, since the call happens synchronously on the execute thread rather than possibly context switching in enqueue (i assume that's how enqueue works?). The main disadvantage would be that it could break clients who are depending on that enqueue behavior.

Anyway, that's something more to think about for the future. It's your call as to whether the intercept should just go inside the create here, so let me know and I'll make the change. :)